internal/cmd/controller/gitops/reconciler/gitjob_controller.go

```diff
@@ -141,8 +144,9 @@ func (r *GitJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 
 	// Restrictions / Overrides, gitrepo reconciler is responsible for setting error in status
 	oldStatus := gitrepo.Status.DeepCopy()
-	gitrepo, err := grutil.AuthorizeAndAssignDefaults(ctx, r.Client, gitrepo)
```

Comment:
Was there an issue with overwriting `gitrepo` here?

Reply:
Yes, as we're using the `gitrepo` in the later Event call... `AuthorizeAndAssignDefaults` returns `nil` when something goes wrong. 
Anyway, when that happens we don't do anything else with that `gitrepo` because we return, but we need it in order to call `Event` and the object (`gitrepo` in this case, is the first parameter)

Reply:
Ah that seems obvious now :) Thanks for clarifying.